### PR TITLE
feat: add randomness support

### PR DESCRIPTION
**Pull request by heueristik**
Adds randomness types, builtins, and syntax aliases.

### CODE REVIEW

File: Anoma/Resource/Types.juvix [hunk 1]
```diff
@@ -26,12 +31,15 @@ type NullifierKeyCommitment :=
...
+--- NOTE: This should be a number having an at most negligible chance of repeating, e.g., a pseudo-random number.
+type Nonce := mkNonce@{unNonce : AnomaEncoded};
 
 --- A fixed-size data type encoding a randomness seed.
 --- NOTE: This seed provides pseudo randomness and cannot be expected to provide true randomness.
-type RandSeed := mkRandSeed@{unRandSeed : Nat};
+type RandSeed := mkRandSeed@{unRandSeed : AnomaEncoded};
+
+--- The size of bytes.
+randSeedBytesSize : AnomaEncoded := 32;
```
Comment (paulcadman) [inline Anoma/Resource/Types.juvix]: It seems like `randSeedBytesSize` should have type `Nat` (I know they're synonyms but `AnomaEncoded` means the encoded bytes).

File: Anoma/Resource/Types.juvix [hunk 2]
```diff
@@ -183,6 +194,8 @@ module NonceInternal;
 end;
 
 module RandSeedInternal;
+  RandSeed-FixedSize : FixedSize RandSeed := mkFixedSize 32;
```
Comment (paulcadman) [inline Anoma/Resource/Types.juvix]: You could use `randSeedBytesSize` instead of `32` here.

Comment (paulcadman) [reply Anoma/Resource/Types.juvix]: @heueristik The fixed size trait has been removed?

File: Anoma/Builtin/ByteArray.juvix [hunk 3]
```diff
@@ -12,20 +12,21 @@ axiom mkByteArray : List Byte -> ByteArray;
 builtin bytearray-length
 axiom length : ByteArray -> Nat;
 
--- TODO: Everything below needs to be removed.
-syntax alias AnomaContents := Nat;
+-- TODO Should be moved into base.
+-- TODO Should we use `AnomaEncoded : Type := Nat;` here?
+syntax alias AnomaEncoded := Nat;
```
Comment (heueristik) [inline Anoma/Builtin/ByteArray.juvix]: WDYT @paulcadman ?

File: Anoma/Resource/Types.juvix (hunk 2 above)
Comment (heueristik) [reply Anoma/Resource/Types.juvix]: Ah, right. This variable should be deleted anyway. I've added [`Primitives.FixedSize` on `main` already](https://github.com/anoma/juvix-arm-specs/blob/main/Anoma/Primitives/FixedSize.juvix) (after seeing Yulia's new specs)

Comment (heueristik) [reply Anoma/Resource/Types.juvix]: Removed in e82c5a532f35c9c67e1a7fe9d0f4ec8dd1c86617

File: Anoma/Builtin/ByteArray.juvix (hunk 3 above)
Comment (paulcadman) [reply Anoma/Builtin/ByteArray.juvix]: discussed in slack, we came up with `AnomaByteArray` as an alternative.

**Pull request merged**